### PR TITLE
DOC: More examples  comparison with sql

### PR DESCRIPTION
**Pull request by maxu777**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

added the following examples:
- Top N rows with offset
- Top N rows per group
- ROW_NUMBER() OVER(PARTITION BY day ORDER BY total_bill DESC)
- UPDATE
- DELETE

### CODE REVIEW

**Comment by jreback** [comment]
pls don't close original PR's and open new ones, just push to the SAME one.

**Comment by jreback** [comment]
http://pandas.pydata.org/pandas-docs/stable/contributing.html

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,554 @@
+Contributing to pandas
+======================
+
+Where to start?
```
Comment (jreback) [inline CONTRIBUTING.md]: this is not part of your PR, pls remove

**Comment by maxu777** [comment]
sorry, i'm new to GitHub stuff - have to learn how to deal with that

**Comment by maxu777** [comment]
@jreback, how can i remove it?

**Comment by jreback** [comment]
just rebase on master

``git rebase -i origin/master``

**Comment by maxu777** [comment]
i either have this error message:

[pandas_dev] C:\Users\Max\Anaconda3\envs\pandas-maxu777\doc>git rebase -i origin/master

It seems that there is already a rebase-merge directory, and
I wonder if you are in the middle of another rebase.  If that is the
case, please try
        git rebase (--continue | --abort | --skip)
If that is not the case, please
        rm -fr "C:/Users/Max/Anaconda3/envs/pandas-maxu777/.git/rebase-merge"
and run me again.  I am stopping in case you still have something
valuable there.


or that:

[pandas_dev] C:\Users\Max\Anaconda3\envs\pandas-maxu777\doc>git commit -m "restored original CONTRIBUTING.md"
interactive rebase in progress; onto 2267bd3
fatal: Could not open file .git/rebase-merge/done for reading: No such file or directory

i'm giving up - unfortunately i don't have that much time to play with `git`
:(

**Comment by jreback** [comment]
``git rebase --abort``

then try again

**Comment by jreback** [comment]
http://stackoverflow.com/questions/2047465/how-can-i-delete-a-file-from-git-repo

you can also do (you need to do the abort though)

``git rm contributing.MD --cached``

**Comment by maxu777** [comment]
[pandas_dev] C:\Users\Max\Anaconda3\envs\pandas-maxu777\doc>git push origin more_examples__comparison_with_sql -f
Warning: Permanently added the RSA host key for IP address '192.30.252.122' to the list of known hosts.
Everything up-to-date

**Comment by maxu777** [comment]
@jreback, i've pushed it now - is it OK now?

**Comment by jreback** [comment]
not sure what you have pushed here, but doesn't look like its updated according to my comments.

**Comment by maxu777** [comment]
@jreback, is it OK now?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -109,9 +109,9 @@ For a python 3 environment:
 
     conda create -n pandas_dev python=3 --file ci/requirements_dev.txt
```
Comment (jreback) [inline CONTRIBUTING.md]: this file should be removed, but if you can't i will do on merge

**Comment by jreback** [comment]
any comments from those SQL inclined

File: doc/source/comparison_with_sql.rst [hunk 3]
```diff
@@ -372,10 +372,98 @@ In pandas, you can use :meth:`~pandas.concat` in conjunction with
...
+        RANK() OVER(PARTITION BY day ORDER BY total_bill DESC) AS rnk
+      FROM tips t
+    )
+    WHERE rnk < 3
+    ORDER BY day, rn;
+
+.. ipython:: python
+
+    tips['rnk_min'] = tips.groupby(['day'])['total_bill'].rank(method='min', ascending=False)
+    tips.loc[tips['rnk_min'] < 3].sort_values(['day','rnk_min'])
```
Comment (jreback) [inline doc/source/comparison_with_sql.rst]: these will already be sorted, no need for ``.sort_values``

File: doc/source/comparison_with_sql.rst [hunk 4]
```diff
@@ -372,10 +372,98 @@ In pandas, you can use :meth:`~pandas.concat` in conjunction with
...
+        t.*,
+        ROW_NUMBER() OVER(PARTITION BY day ORDER BY total_bill DESC) AS rn
+      FROM tips t
+    )
+    WHERE rn <= 3
+    ORDER BY day, rn;
+
+.. ipython:: python
+
+    tips.sort_values(['total_bill'], ascending=False).groupby('sex').head(3)
```
Comment (jreback) [inline doc/source/comparison_with_sql.rst]: show these with chained method syntax e.g.

```
(tips.sort_values(['total_bill'], ascending=False)
       .groupby('sex')
       .head(3)
)

File: doc/source/comparison_with_sql.rst [hunk 5]
```diff
@@ -372,10 +372,98 @@ In pandas, you can use :meth:`~pandas.concat` in conjunction with
...
+    tips['rn'] = tips.sort_values(['total_bill'], ascending=False) \
+                     .groupby(['day']) \
+                     .cumcount() + 1
+    tips.loc[tips['rn'] < 3].sort_values(['day','rn'])
+
+the same using `rank(method='first')` function
+
+.. ipython:: python
+
+    tips['rnk'] = tips.groupby(['day'])['total_bill'].rank(method='first', ascending=False)
```
Comment (jreback) [inline doc/source/comparison_with_sql.rst]: use ``.query`` like above

File: doc/source/comparison_with_sql.rst (hunk 3 above)
Comment (jreback) [inline doc/source/comparison_with_sql.rst]: same

**Comment by TomAugspurger** [comment]
+1 on the SQL docs.

File: doc/source/comparison_with_sql.rst [hunk 6]
```diff
@@ -372,10 +372,99 @@ In pandas, you can use :meth:`~pandas.concat` in conjunction with
...
+.. ipython:: python
+
+    tips['rn'] = (tips.sort_values(['total_bill'], ascending=False)
+                      .groupby(['day'])
+                      .cumcount() + 1
+                 )
+    tips.loc[tips['rn'] < 3].sort_values(['day','rn'])
+
+the same using `rank(method='first')` function
+
```
Comment (jreback) [inline doc/source/comparison_with_sql.rst]:
```
(tips.assign(rn=tips.sort_values(['total_bill'], ascending=False)
                    .groupby(['day'])
                    .cumcount() + 1)
     .query('rn < 3')
)

File: doc/source/comparison_with_sql.rst [hunk 7]
```diff
@@ -372,10 +372,99 @@ In pandas, you can use :meth:`~pandas.concat` in conjunction with
...
+      FROM tips t
+    )
+    WHERE rnk < 3
+    ORDER BY day, rn;
+
+.. ipython:: python
+
+    tips['rnk_min'] = (tips.groupby(['day'])['total_bill']
+                           .rank(method='min', ascending=False)
+                      )
```
Comment (jreback) [inline doc/source/comparison_with_sql.rst]: same here

**Comment by maxu777** [comment]
@jreback, thanks for the hint: `assign().query()` - i wasn't aware of `assign()` function

**Comment by jreback** [comment]
thanks!

FYI, I think somehow the ``CONTRIBUTING.md`` file got moved to ``.github/CONTRIBUTING.md``; that's why it showed up. I removed that change.

**Pull request closed without merging**